### PR TITLE
ref: extract implementations of other classes from SentryProfiler.mm

### PR DESCRIPTION
**Pull request by armcknight**
I forgot to move the implementation of SentryProfilerState in #3124 

#skip-changelog

### CODE REVIEW

File: Sentry.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -3086,37 +3091,40 @@
 		8405A517279906EF001B38A1 /* Profiling */ = {
 			isa = PBXGroup;
 			children = (
-				84354E0F29BF944900CDBB8B /* SentryProfileTimeseries.h */,
```
Comment (armcknight) [inline Sentry.xcodeproj/project.pbxproj:3089]: I sorted the contents of the Profiling group by name.

File: Sources/Sentry/include/SentryProfilerState.h [hunk 2]
```diff
@@ -6,9 +6,22 @@
...
+ * This function would return:
+ * @code -[UIFieldEditor _fullContentInsetsFromFonts] @endcode
+ * @note If the format does not match the expected format, this returns the input string.
+ */
+NSString *parseBacktraceSymbolsFunctionName(const char *symbol);
+
 @class SentrySample;
 
-@interface SentryProfilingMutableState : NSObject
+@interface SentryProfilerMutableState : NSObject
```
Comment (armcknight) [inline Sources/Sentry/include/SentryProfilerState.h:24]: I'd added this to a file called SentryProfilerState but the interface name was SentryProfilingState. Aligning them both to reduce confusion.

**Review by brustolin — APPROVED** [review]
LGTM

**Review by philipphofmann — APPROVED** [review]
LGTM

**Pull request merged**